pandera/schemas.py

```diff
@@ -1969,23 +1969,40 @@ def validate(
                     ),
                 )
 
-        if self._dtype is not None and (
-            not self._dtype.check(pandas_engine.Engine.dtype(series.dtype))
-        ):
-            msg = (
-                f"expected series '{series.name}' to have type {self._dtype}, "
-                + f"got {series.dtype}"
-            )
-            error_handler.collect_error(
-                "wrong_dtype",
-                errors.SchemaError(
-                    self,
-                    check_obj,
-                    msg,
-                    failure_cases=scalar_failure_case(str(series.dtype)),
-                    check=f"dtype('{self.dtype}')",
-                ),
+        if self._dtype is not None:
+            failure_cases = None
+
+            check_output = self._dtype.check(
+                pandas_engine.Engine.dtype(series.dtype), series
             )
+            if check_output is False:
+                failure_cases = scalar_failure_case(str(series.dtype))
+                msg = (
+                    f"expected series '{series.name}' to have type {self._dtype}, "
+                    + f"got {series.dtype}"
+                )
+            else:
```

Comment:
this needs to handle the case where `check_output` is an empty series

for example, this change makes tests pass:
```python
            if check_output is False:
                failure_cases = scalar_failure_case(str(series.dtype))
                msg = (
                    f"expected series '{series.name}' to have type {self._dtype}, "
                    + f"got {series.dtype}"
                )
            elif not isinstance(check_output, bool) and not all(check_output):
                _, failure_cases = check_utils.prepare_series_check_output(
                    series, check_output
                )
                reshaped_failure_cases = reshape_failure_cases(failure_cases)
                msg = (
                    f"expected series '{series.name}' to have type {self._dtype}:\n"
                    f"failure cases:\n{reshaped_failure_cases}"
                )
```